impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -333,13 +334,28 @@ class AllocatedTextureSourceVK final : public TextureSourceVK {
       return;
     }
     image_view_ = std::move(image_view);
-
     is_valid_ = true;
   }
 
   ~AllocatedTextureSourceVK() {
-    TRACE_EVENT0("impeller", "DestroyDeviceTexture");
     image_view_.reset();
+    auto context = context_.lock();
+    if (context && ContextVK::Cast(*context).IsOnRasterThread() && image_) {
+      ContextVK::Cast(*context).GetConcurrentWorkerTaskRunner()->PostTask(
```

Comment:
Would it be simpler to add some API to `ConcurrentTaskRunner::RunNowOrPostTask`?

It would check if the message loop is the concurrent loop or not, if it is run now, else post task.